api/src/main/java/org/openmrs/validator/DiagnosisValidator.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.validator;
+
+import org.openmrs.Diagnosis;
+import org.openmrs.api.APIException;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.validation.Validator;
+
+public class DiagnosisValidator implements Validator {
```

Comment:
As for the validator, did you take a look at the validators in the link i gave on the parent ticket?

Reply:
Yeah, I did but did not fully grasp it all at once. I am looking at it again and creating a better Validator.

Reply:
Did you just forget to make some changes to this class? One of you has already done this validator so well.